docs/assembly.rst

```diff
@@ -394,6 +394,17 @@ use ``x_slot``, and to retrieve the byte-offset you use ``x_offset``.
 
 Local Solidity variables are available for assignments, for example:
 
+.. warning::
+    If you access variables of a type that spans less than 256 bits
+    (for example ``uint64``, ``address``, ``bytes16`` or ``byte``),
+    you cannot make any assumptions about bits not part of the
+    encoding of the type. Especially, do not assume them to be zero.
+    To be safe, always clear the data properly before you use it
+    in a context where this is important:
+    ``uint32 x = f(); assembly { x := and(x, 0xffffffff) /* now use x */ }``
+    To clean signed types, you can use the ``signextend`` opcode:
+    ``assembly { signextend(0, x) }``
```

Comment:
I'm not sure if this is right: 

```
signextend(i, x): sign extend from (i*8+7)th bit counting from least significant
```

so signextend(0,x) will sign extend from the 7th bit from LSB. What happens if you have a signed 32 bit integer (int32). Wouldn't it change the value itself?

Perhaps, this is what you meant?

```
assembly { signextend(<bit_width_of_x>, x) }
```

Reply:
@Meshugah Purely for readability, I think this warning block is better under the code block.